lecture-notes/2-vcs.md

```diff
@@ -1429,7 +1429,111 @@ even works outside a Git repository!
 
 #### Undoing mistakes
 
-TODO: Using `git revert` to undo an old change.
+Viewing past changes is well and good, and it'll often help you figure out why
+the code you've been editing has mysteriously stopped working. But once you've
+figured out the problem, what can you do about it? You could hope that your text
+editor's undo history is long enough; you could copy and paste individual lines
+from the output of `git diff` or `git show`. Or, you could use one of several
+Git subcommands, detailed in the next few sections, that apply pieces of your
+Git history to files in the working tree.
+
+`git revert` is the most intuitive of these commands. Given a revision
+parameter, `git revert` generates a new commit that undoes the changes of just
+that commit, preserving changes from other intervening commits:
```

Comment:
you say "that commit" but earlier you said "revision parameter" - can you maybe introduce a "variable" `C` or something

Reply:
it might help with like "it views the commit as a diff and tries to apply that diff to the current commit"